apps/coordinator/ts/scheduler/scheduler.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { Body, Controller, HttpException, HttpStatus, Logger, Post } from "@nestjs/common";
+import { ApiResponse, ApiTags } from "@nestjs/swagger";
+
+import { IIdentityScheduledPoll } from "../redis/types";
+
+import { IdentityScheduledPollDto, SchedulePollWithSignerDto } from "./dto";
+import { SchedulerService } from "./scheduler.service";
+import { IIsPollScheduledResponse, ISchedulePollWithSignerArgs } from "./types";
+
+@ApiTags("v1/scheduler")
+@Controller("v1/scheduler")
+export class SchedulerController {
+  /**
+   * Logger
+   */
+  private readonly logger = new Logger(SchedulerController.name);
+
+  /**
+   * Initialize SchedulerController
+   *
+   * @param schedulerService - scheduler service
+   */
+  constructor(private readonly schedulerService: SchedulerService) {}
+
+  /**
+   * Register poll for finalization
+   */
+  @ApiResponse({ status: HttpStatus.OK, description: "Poll finalization has been scheduled" })
+  @ApiResponse({ status: HttpStatus.BAD_REQUEST, description: "BadRequest" })
+  @Post("register")
+  async register(@Body() args: SchedulePollWithSignerDto): Promise<IIsPollScheduledResponse> {
+    const schedulePollWithSignerArgs = {
+      ...args,
+      pollId: args.pollId.toString(),
+      endDate: 0,
+      merged: false,
+      proofsGenerated: false,
+    } as ISchedulePollWithSignerArgs;
+
+    return this.schedulerService.registerPoll(schedulePollWithSignerArgs).catch((error: Error) => {
```

Comment:
```suggestion
    return this.schedulerService.registerPoll(schedulePollWithSignerArgs).catch((error) => {
```

Reply:
Oh, didn't know that it doesn't infer the type automatically. If so, let's stick to the first option, it's better than explicit casting.